faunadb/client.go

```diff
@@ -8,16 +8,45 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"time"
 )
 
-const DefaultEndpoint = "https://cloud.faunadb.com"
+const (
+	DefaultEndpoint = "https://cloud.faunadb.com"
+	requestTimeout  = 60 * time.Second
+)
 
 var resource = ObjKey("resource")
 
+type ClientConfig func(*FaunaClient)
+
+func Endpoint(url string) ClientConfig    { return func(cli *FaunaClient) { cli.endpoint = url } }
+func HTTP(http *http.Client) ClientConfig { return func(cli *FaunaClient) { cli.http = http } }
+
 type FaunaClient struct {
-	Secret   string
-	Endpoint string
-	HTTP     http.Client
+	basicAuth string
+	endpoint  string
+	http      *http.Client
+}
+
+func NewFaunaClient(secret string, configs ...ClientConfig) *FaunaClient {
+	client := &FaunaClient{basicAuth: basicAuth(secret)}
+
+	for _, config := range configs {
+		config(client)
+	}
+
+	if client.endpoint == "" {
```

Comment:
Just a thought: wouldn't be better to check for nullability and settings defaults on `Endpoint()` and `HTTP()` functions? That way you could make that function more clean